### DISCUSSION TITLE
patterns/dependency-injection/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#80] [Announcements]
# patterns/dependency-injection/

Intent Dependency Injection is a software design pattern in which one or more dependencies (or services) are injected, or passed by reference, into a dependent object (or client) and are made part of the client's state. The pattern separates the creation of a client's dependencies from its own behavior, which allows program designs to be loosely coupled and to follow the inversion of control and single responsibility principles.

https://java-design-patterns.com/patterns/dependency-injection/

**Comment by salamanetwork** [comment]
Amazing, Awesome!
